SFT_README.md

```diff
@@ -0,0 +1,114 @@
+# Supervised Fine-tuning with Modalities
+
+Currently supported are Instruction-tuning and Low-rank Adaption (LorA), as explained in more detail next.
+
+## Instruction-tuning
+* entry point to prepare data
+* jinja2 templates
+* The b_include_to_loss_token, e_include_to_loss_token are required to be part of each chat template for proper loss masking!
+* hash to connect files
+ 
+* truncation, padding
+* re-use last target
+
+### Create Prompts from Conversations
+To prepare the instruction-tuning data we created a new entry point `apply_chat_template`, which requires a [configuration file](./config_files/data_preparation/apply_chat_template_config.yaml). Wihtin it we define:
+* the path to instruction-tuning dataset as a JSONL file wereas each line contains a structured conversation as an array of dictionaries.
+* A [jinja2](https://jinja.palletsprojects.com/en/3.1.x/) chat template which defines the rules how to glue `chat_template_data` and the data within the JSONL together to one `chat` string.
+
+As part of the `chat_template_data`, we require the special tokens `b_include_to_loss_token` and `e_include_to_loss_token`. 
+> ❗ You should choose sequences which are tokenized into a single token and will not appear in the assistant utterances of the instruction-tuning data!
+
+They are used to mark the begin and end of the assistant turns, as we need to include only tokens between those into the loss computation during instruction-tuning with modalities.
+
+```yaml
+chat_template_data:
+  ...
+  special_tokens:
+      b_include_to_loss_token: ^
+      e_include_to_loss_token: $
+```
+
+Run the `apply_chat_template` entry point with:
+```bash
+modalities data apply_chat_template --config_file_path config_files/data_preparation/apply_chat_template_config.yaml
+```
+
+This will create two files
+1. The new JSONL file with a new attribute `chat` containing the conversations e.g. `lorem_ipsum_sft_converted.aadd295.jsonl`
+2. The config used to generate the `chat` e.g. `sft_chat_template_config.aadd295.yaml`
+
+> Both files names contain the first 7 symbols of the hash of the config file, to group files which belong together!
+
+### Create idx and pbin files
+Before continuing with the instruction-tuning you need to index the created JSONL and convert it to a packed data file.
```

Comment:
It's not really "packed" (only tokenized and padded sequences right?) and we already have the misnomer in other parts of the code. Maybe say tokenized binary file?